cmd/snap/cmd_run.go

```diff
@@ -1225,6 +1347,13 @@ func (x *cmdRun) runSnapConfine(info *snap.Info, securityTag, snapApp, hook stri
 			logger.Debugf("snap refreshes will not be postponed by this process")
 		}
 	}
+
+	if beforeExec != nil {
+		if err := beforeExec(); err != nil {
```

Comment:
@pedronis should we be retrying after transient cgroups are created? would we be creating multiple cgroups for the same process, hence leaking cgroups?

Reply:
as I said we could explore removing the cgroup if we need to retry, related to the leaking we could also remember that we created it already and not doing it again